File: python/2587.py
int_list = []
for i in range(5):
    k = int(input())
    flag = False
    for j in range(i):
        if k > int_list[j]:
            continue
        elif k <= int_list[j]:
            flag = True
            break

    if i == 0 or not flag:
        int_list.append(k)
    elif flag:
        int_list.insert(j, k)

mid = int_list[len(int_list) // 2]
avg = sum(int_list) // len(int_list)
print(avg)
print(mid)

File: rust/src/bin/2587.rs
use std::io;

fn main() {
  let mut i = String::new();
  let mut arr: Vec<usize> = Vec::new();

  
  
  for _ in 0..5 {
    io::stdin()
      .read_line(&mut i)
      .expect("input error");

    let k: usize = i.trim().parse().expect("NaN");
    arr.push(k);
    i.clear();
  }  
  arr.sort();
  let sum = arr.iter().fold(0, |acc, x| acc + x);
  let avg = sum / arr.len();
  println!("{}\n{}", avg,arr[arr.len()/2]);

}